manticore/core/smtlib/solver.py

```diff
@@ -388,39 +355,35 @@ def _getvalue(self, expression):
         """
         if not issymbolic(expression):
             return expression
-        assert isinstance(expression, Variable)
 
         if isinstance(expression, Array):
             result = bytearray()
             for c in expression:
                 expression_str = translate_to_smtlib(c)
-                self._send("(get-value (%s))" % expression_str)
-                response = self._recv()
-                result.append(int("0x{:s}".format(response.split(expression_str)[1][3:-2]), 16))
+                # self._smtlib.send(f"(get-value ({expression_str}))")
+                # response = self._smtlib.recv().split(expression_str)[1].strip(")")
+                # pattern, base = self._get_value_fmt
+                # m = pattern.match(response)
+                # expr, value = m.group("expr"), m.group("value")
+                # result.append(int(value, base))
+                result.append(self.__getvalue_bv(expression_str))
             return bytes(result)
         else:
-            self._send("(get-value (%s))" % expression.name)
-            ret = self._recv()
-            assert ret.startswith("((") and ret.endswith("))"), ret
-
-            if isinstance(expression, Bool):
-                return {"true": True, "false": False}[ret[2:-2].split(" ")[1]]
-            elif isinstance(expression, BitVec):
-                pattern, base = self._get_value_fmt
-                m = pattern.match(ret)
-                expr, value = m.group("expr"), m.group("value")
-                return int(value, base)
+            if isinstance(expression, BoolVariable):
+                return self.__getvalue_bool(expression.name)
+            elif isinstance(expression, BitVecVariable):
+                return self.__getvalue_bv(expression.name)
 
-        raise NotImplementedError("_getvalue only implemented for Bool and BitVec")
+        raise NotImplementedError("_getvalue only implemented for Bool, BitVec and Array")
```

Comment:
Would be helpful to show the type of what was passed.
```suggestion
        raise NotImplementedError(f"_getvalue only implemented for Bool, BitVec and Array. Got {type(expression)}")
```